src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const arrayStates = [];
+
+  for (const key of actions) {
+    switch (key.type) {
+      case 'addProperties':
+        Object.assign(newState, key.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const action of key.keysToRemove) {
+          delete newState[action];
+        }
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
+    arrayStates.push({ ...newState });
```

Comment:
Pushing the copy of the state object to the `arrayStates` should be done outside of the `switch` block to follow the DRY (Don't Repeat Yourself) principle. This action is common to all cases and does not need to be repeated within each case.